resources/contents/blog/en/readme-driven-development.md

```diff
@@ -0,0 +1,75 @@
+
+**Write the Readme first.**
+
+Read it again. Write the Readme first. With first, it means before anything. It means before you the first line of code, it means before the first test and it even means before you setup the project.
+
+And read it again. It says Readme. It does not say documentation, it does not say manual and it does not say specifications. Just Readme.
+
+**Write the Readme first.**
+
+That's all. This rule will help to avoid that pieces of software becomes useless once it is finished.
+
+## Implementing Readme Driven Development
+
+But what does it exactly mean to write the Readme first? It is pretty straight forward. Just before you start working, take the time and start writing a simple file that explains what it is all about.
+
+Start with a title and a subtitle that makes it directly clear what the topic is. Then continue with a small paragraph explaining what the code you write intends to do and what it is going to solve.
+
+Then follow with some *Getting Started* information, so that the user of this piece of software will be able to set things up and configure things if needed.
+
+Next explain the usage of what you are writing. Describe the most important parts of the public interface of you software and how to use them.
+
+Now do this every single time whenever the code needs some changes. Update the Readme to the state your code will be after applying the changes before you actually change the code.
+
+## The power of Readme Driven Development
+
+You might wonder why one should spend that much time upfront on a Readme. Once you start on a project, you will always have some loose coupled ideas on how you want to design the software. You have some experience about software design patterns and you have already solved some similar problems and so you basically know what you want to do.
+
+The problem about start coding immediately is that you will miss the bigger picture. You won't notice that one feature might interfere with another until you start implementing the second one, so you end up re-factoring everything and maybe even have to reconsider the whole design of the software.
+
+To make other people use your software, you have to make sure that people understand and accept the thoughts you initially have when you start. You have to make sure that you are actually understanding tbeir needs, why they actually consider using your software. By writing down the Readme, you can easily discuss ideas without spending too much time on working in the wrong direction.
+
+But it can also be helpful for yourself to verify if the idea you had actually makes sense. Writing the Readme and then re-reading it the day after is the best verification to see if you are doing exactly what you did. And if not, you haven't spent too much time to start over, right?
```

Comment:
```suggestion
But it can also be helpful for yourself to verify if the idea you had actually makes sense. Writing the Readme and then re-reading it the day after is the best verification to see if you are doing exactly what you wanted to do. And if not, you haven't spent too much time to start over, right?
```